libs/designsystem/src/lib/components/chart-wip/chart-wip.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+
+      expect(element).toHaveComputedStyle({ height: `${customHeight}px` });
+    });
+
+    it('should be possible to use a string', () => {
+      const customHeight = '600px';
+      const element = spectator.query('.chart-container') as HTMLElement;
+      expect(element).not.toHaveComputedStyle({ height: customHeight });
+
+      spectator.setInput('height', customHeight);
+      spectator.detectChanges();
+
+      expect(element).toHaveComputedStyle({ height: customHeight });
+    });
+  });
+
+  describe('when changes occur to', () => {
+    const scenarios = {
+      customOptions: {
+        updateFn: 'updateCustomOptions',
+        newValue: { onClick: () => console.log('testing') },
+      },
+      data: { updateFn: 'updateData', newValue: [1, 2, 3] },
+      dataLabels: { updateFn: 'updateDataLabels', newValue: ['one', 'two', 'three'] },
+      type: { updateFn: 'updateType', newValue: 'bar' },
+      annotations: { updateFn: 'updateAnnotations', newValue: [{ type: 'line' }] },
+    };
+
+    Object.entries(scenarios).forEach(([property, { updateFn, newValue }]) => {
+      describe(`${property}`, () => {
+        it(`should only update ${property}`, () => {
+          const updateFnSpy = spyOn<any>(component, updateFn);
+
+          spectator.setInput(property as any, newValue);
+
+          expect(updateFnSpy).toHaveBeenCalledTimes(1);
+        });
+
+        it('should redraw once', () => {
+          const redrawChartSpy = spyOn<any>(component, 'redrawChart');
+
+          spectator.setInput(property as any, newValue);
+
+          expect(redrawChartSpy).toHaveBeenCalledTimes(1);
+        });
+      });
+    });
+
+    describe('multiple chartJS related input properties at the same time', () => {
+      it('should update all of the properties once', () => {
+        const updateFnSpies = [
+          spyOn<any>(component, 'updateData'),
+          spyOn<any>(component, 'updateDataLabels'),
+          spyOn<any>(component, 'updateType'),
+          spyOn<any>(component, 'updateCustomOptions'),
+        ];
+
+        spectator.setInput({
+          data: [1, 2, 3],
+          dataLabels: ['one', 'two', 'three'],
+          type: 'bar',
+          customOptions: { onClick: () => console.log('testing') },
+        });
+
+        updateFnSpies.forEach((updateFnSpy) => {
+          expect(updateFnSpy).toHaveBeenCalledTimes(1);
+        });
+      });
+
+      it('should redraw chart once', () => {
+        const redrawChartSpy = spyOn<any>(component, 'redrawChart');
+
+        spectator.setInput({
+          data: [1, 2, 3],
+          dataLabels: ['one', 'two', 'three'],
+          type: 'bar',
+          customOptions: { onClick: () => console.log('testing') },
+        });
+
+        expect(redrawChartSpy).toHaveBeenCalledTimes(1);
+      });
+    });
+  });
+
+  describe('when "type" is not provided', () => {
+    it('should default to correct type', () => {
+      expect(component.type).toBe('column');
+    });
+  });
```

Comment:
Could we move this closer to the top? I think it fits well with `should have correct default height`-tests and the ones like it.

Reply:
Has been done :D